tests/system/python/packages/test_gcp_gateway.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test GCP Gateway plugin
+
+Prerequisite: This test expects gcp certificates in $FOGLAMP_ROOT/tests/system/python/packages/data/gcp directory.
+              Also export the GOOGLE_APPLICATION_CREDENTIALS variable to the JSON file which has details of service account.
+              Example: export GOOGLE_APPLICATION_CREDENTIALS="[PATH]"
+"""
+
+import os
+import subprocess
+import http.client
+import json
+import time
+import pytest
+from pathlib import Path
+import utils
+from datetime import datetime
+
+__author__ = "Yash Tatkondawar"
+__copyright__ = "Copyright (c) 2020 Dianomic Systems Inc."
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+task_name = "gcp-gateway"
+north_plugin = "GCP-Gateway"
+# This  gives the path of directory where FogLAMP is cloned. test_file < packages < python < system < tests < ROOT
+PROJECT_ROOT = Path(__file__).parent.parent.parent.parent.parent
+SCRIPTS_DIR_ROOT = "{}/tests/system/python/packages/data/".format(PROJECT_ROOT)
+FOGLAMP_ROOT = os.environ.get('FOGLAMP_ROOT')
+CERTS_DIR = "{}/gcp".format(SCRIPTS_DIR_ROOT)
+FOGLAMP_CERTS_DIR = "{}/data/etc/certs/".format(FOGLAMP_ROOT)
+
+# [[ GCP project specific configuration ]]
+project_id = "nomadic-groove-264509"
+registry_id = "fl-nerd--registry"
+device_id = "fl-nerd-gateway"
+
+subscription_name = "my-subscription"
+# [[ GCP project specific configuration ]]
+
+@pytest.fixture
+def reset_foglamp(wait_time):
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./reset"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "reset package script failed!"
+
+    # Wait for foglamp server to start
+    time.sleep(wait_time)
+
+
+@pytest.fixture
+def remove_and_add_pkgs(package_build_version):
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./remove"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "remove package script failed!"
+
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package/ && ./setup {}"
+                       .format(PROJECT_ROOT, package_build_version)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "setup package script failed"
+
+    try:
+        subprocess.run(["sudo apt install -y foglamp-north-gcp-gateway foglamp-south-sinusoid"], shell=True, check=True)
```

Comment:
We should make it `foglamp-north-gcp` from `nightly/buster/armv7l/` (atleast as long as we have it with foglamp/ org; and then think again once its migrated....)